src/Metadata/ClassMetadata.php

```diff
@@ -384,6 +353,48 @@ public function unserialize($str)
         parent::unserialize($parentStr);
     }
 
+    private function setDiscriminatorProperty()
```

Comment:
can you please rename this method using a different name? this is not a setter. something as "handleDiscriminatorMap" ?

Reply:
done